lab1/src/icecream.ts

```diff
@@ -0,0 +1,117 @@
+enum Size {
+    Small = "small",
+    Large = "large"
+}
+
+enum Topping {
+    Chocolate = "chocolate",
+    Caramel = "caramel",
+    Berries = "berries"
+}
+
+enum YesNo {
+    Yes = "yes",
+    No = "no"
+}
+
+const SizeCosts = {
+    [Size.Small]: 10,
+    [Size.Large]: 25
+}
+
+const ToppingCosts = {
+    [Topping.Chocolate]: 5,
+    [Topping.Caramel]: 6,
+    [Topping.Berries]: 10
+}
+
+const MarshmallowCost = 5;
+
+function calculateIceCreamCost(size: Size, toppings: Topping[], addMarshmallow: YesNo): number {
+    let cost = SizeCosts[size];
+
+    toppings.forEach((topping) => {
+        cost += ToppingCosts[topping];
+    });
+
+    if (addMarshmallow === YesNo.Yes) {
+        cost += MarshmallowCost;
+    }
+
+    return cost;
+}
+
+function isSize(input: string): input is Size {
+    return Object.values(Size).includes(input as Size);
+}
+
+function isTopping(input: string): input is Topping {
+    return Object.values(Topping).includes(input as Topping);
+}
+
+function isYesNo(input: string): input is YesNo {
+    return Object.values(YesNo).includes(input as YesNo);
+}
+
+let sizeInput: string | null;
+let toppingsInput: string | null;
+let toppingsArray: string[];
+let marshmallowInput: string | null;
+
+while (true) {
```

Comment:
DRY, you have three separate while loops for user inputs (line 61, line 77, line 99). Create validation function for user input.

Reply:
Split the code into getInput() func with a while loop and validation functions